### ISSUE TITLE
p,q and d are incorrect in IBM_February2024.pdf

### ISSUE THREAD

**Issue by ChrisNerd** [jbayerl/IBM-Ponder-This-Puzzles#1]
For IBM_February2024.pdf, I was trying to follow how you derived the equations for P0, especially without Markov chains. I'm only partially understanding it.

However, I found it weird that your answers for p, q and d differ between Part 1 and Part 2. I reran your code and you correctly get the integer values for prime_count = 114399, evenNonPrime=116001, and len(total)=460800.
Without messing with the numerical precision in python at all, these ratios are represented exactly.
p=0.24826171875
q=0.25173828125
d=0.5
Furthermore, Part 2 only swaps d and q.
I'm curious as to how you ended up with the seemingly far more accurate values.

And, btw, you mentioned "the following is not pretty but does its job" with your 6 embedded for loops.
You might be interested in my functional one-liner.
```
    dice = [
        [0] + [1] * 4,
        [0] + [1] * 6,
        [0] + [1] * 8,
        [1] * 10,
        [0] + [1] * 12,
        [0] + [1] * 20
    ]
    output_distribution = functools.reduce(np.convolve, dice)
    alice_wins = [count for (ind, count) in enumerate(output_distribution) if sympy.isprime(ind)]
```
https://github.com/ChrisNerd/ibmponderthis/blob/main/ponderfeb2024/main.py

**Comment by jbayerl** [comment]
You are right, I used decimal (https://docs.python.org/3/library/decimal.html) and it messed with the p,q,d values a bit. Thanks for your code, that is MUCH more elegant! Let me know where the calculations is not clear; I will try to elaborate

**Comment by ChrisNerd** [comment]
Interesting about the Decimal library. I'm surprised that created an error instead of avoiding one.

So, if you wouldn't mind, could you explain how you created the equation
P_0 = d P_0 + p P_1 + qd P_0 + qp P_1 +... + q^(N-1)d P_0 + q^(N-1)p P1 ?

I recognized that each term corresponds to a "path" that starts from the "tied node" and ends at either the tied node or the "Alice 1-streak" node. But I don't see why those terms should sum to P_0.

Is there a concept from graph theory you employed, or did you just reason this equation into being?

Thanks!

**Comment by jbayerl** [comment]
Think of it like this, P_0 is the probability for Alice to win the game (i.e. win 13 consecutive matches before Bob does). We will calculate P_0 as the sum of all possible ways for Alice to win. First way: starts with a draw, then the probability to win is the same as at the start (i.e. P_0), second way: win for Alice times probability of Alice winning the game after she already scored one point (i.e. P_1), etc... . Reasoning like this we exhaust the entire space of possible ways for Alice to win without any intersections -> P_0

If you rather want to look at it in terms of Markov chains (ultimately the same approach) I suggest you to look at Christian Remling's answer here https://mathoverflow.net/questions/463321/winning-game-probability.

**Comment by ChrisNerd** [comment]
That cleared it up after I went through your logic and there are two good derivations in that link. Thanks!
